server/github.go

```diff
@@ -158,22 +159,36 @@ func (s *Server) getFilenamesInPullRequest(pr *model.PullRequest) ([]string, err
 	return filenames, nil
 }
 
-func (s *Server) checkUserPermission(user, repoOwner string) bool {
-	_, resp, err := s.GithubClient.Organizations.GetOrgMembership(context.Background(), user, repoOwner)
-	if resp.StatusCode == 404 {
-		mlog.Info("User is not part of the ORG", mlog.String("User", user))
-		return false
+func (s *Server) getMembers(ctx context.Context) (orgMembers *[]string, err error) {
+	opts := &github.ListMembersOptions{
+		ListOptions: github.ListOptions{},
```

Comment:
Since the default is `all` https://developer.github.com/v3/orgs/members/#members-list, we don't handle pagination.

Reply:
The `all` does mean, if a `filter` is applied. The GitHub REST API does do pagination my default and limits the number of returned elements to 30 (https://developer.github.com/v3/#pagination). I can verify that only 30 users are returned when I run this code locally.

Reply:
I added additional test code to check for this. Everything is returned, but it uncovered a different bug.

Reply:
I don't think a unit test can catch an issue with the behaviour of `go-github` given that they are mocking away exactly that behaviour that causes issues.

Reply:
Please take a look at the following code:

```golang
package main

import (
	"context"
	"log"

	"github.com/google/go-github/github"
)

func main() {
	ctx := context.Background()

	client := github.NewClient(nil)

	opts := &github.ListMembersOptions{
		ListOptions: github.ListOptions{},
	}
	users, _, err := client.Organizations.ListMembers(ctx, "mattermost", opts)
	if err != nil {
		panic(err)
	}

	log.Printf("len(users): %#+v\n", len(users))
}

```

It's basically a shortened version of the code in `mattermod`. It always returns `30` users, because that is the limit for one request by default.